dot/core/service.go

```diff
@@ -550,3 +576,42 @@ func (s *Service) GetMetadata(bhash *common.Hash) ([]byte, error) {
 	rt.SetContextStorage(ts)
 	return rt.Metadata()
 }
+
+// RegisterRuntimeUpdatedChannel function to register chan that is notified when runtime version changes
+func (s *Service) RegisterRuntimeUpdatedChannel(ch chan<- runtime.Version) (uint32, error) {
+	s.runtimeUpdateSubscriptionsLock.Lock()
+	defer s.runtimeUpdateSubscriptionsLock.Unlock()
+
+	if len(s.runtimeUpdateSubscriptions) == 256 {
+		return 0, errors.New("channel limit reached")
+	}
+
+	id := s.generateID()
+
+	s.runtimeUpdateSubscriptions[id] = ch
+	return id, nil
+}
+
+// UnregisterRuntimeUpdatedChannel function to unregister runtime updated channel
+func (s *Service) UnregisterRuntimeUpdatedChannel(id uint32) bool {
+	s.runtimeUpdateSubscriptionsLock.Lock()
+	defer s.runtimeUpdateSubscriptionsLock.Unlock()
+	ch, ok := s.runtimeUpdateSubscriptions[id]
+	if ok {
+		close(ch)
+		delete(s.runtimeUpdateSubscriptions, id)
+		return true
+	}
+	return false
+}
+
+func (s *Service) generateID() uint32 {
+	uuid := uuid.New()
```

Comment:
put this in the for loop to generate a new one each iteration.

Reply:
Oh, of course, now I see.